friendshipper/src-tauri/src/builds/router.rs

```diff
@@ -228,19 +230,49 @@ where
         };
     }
 
-    match fs::create_dir_all(&local_path) {
+    let local_path_clone = local_path.clone();
+    match fs::create_dir_all(&local_path_clone) {
         Ok(_) => {
-            state.longtail.get_archive(
-                &local_path,
-                None,
-                &archive_urls,
-                tx,
-                aws_client.get_credentials().await,
-            )?;
+            let (cancel_tx, mut cancel_rx) = oneshot::channel();
+            state.cancel_tx.write().await.replace(cancel_tx);

```

Comment:
I don't really understand the purpose of this line, could use a comment

Reply:
ah I think I see, basically this is storing the cancel channel into the app state